games/game-nomad.js

```diff
@@ -0,0 +1,540 @@
+				'win04':    'Endgame scene, Alliance victory',
+			};
+
+			const palettes = {
+				'backg':    'backg.pal',
+				'cock1':    'backg.pal',
+				'cock2':    'backg.pal',
+				'cock3':    'backg.pal',
+				'cock4':    'backg.pal',
+				'cock5':    'backg.pal',
+				'crashed':  'backg.pal',
+				'cred0001': 'cred0001.pal',
+				'cred0002': 'cred0001.pal',
+				'cred0003': 'cred0001.pal',
+				'cred0004': 'cred0001.pal',
+				'cred0005': 'cred0001.pal',
+				'cred0006': 'cred0001.pal',
+				'end1a':    'end1a.pal',
+				'end1b':    'end1b.pal',
+				'fixed':    'backg.pal',
+				'getname':  'getname.pal',
+				'korok01':  'korok01.pal',
+				'korok02':  'korok02.pal',
+				'oesi':     'backg.pal',
+				'open08':   'open08.pal',
+				'snow':     'backg.pal',
+				'win01':    'win01.pal',
+				'win03':    'win03.pal',
+				'win04':    'win04.pal',
+			};
+
+			// Depending on the area of the game's code in which a fullscreen image
+			// is used, the file extension (.lbm) is sometimes omitted.
+			// Add it if necessary.
+			let filename = attr.value;
+			if (filename.substr(filename.length - 4) !== '.lbm') {
+				filename = filename + '.lbm';
+			}
+			
+			// Function to extract the raw image file.
+			const fnExtract = () => getFileFromDat(this.datTest, filename).getContent();
+
+			// Function to overwrite the file.
+			const fnReplace = content => {
+				// Replace getContent() with a function that returns the new content.
+				let file = getFileFromDat(this.datTest, filename);
+				file.getContent = () => content;
+				file.nativeSize = content.length;
+				file.diskSize = undefined; // don't know until written
+			};
+
+			fullscreen[`fullscreen.${index}`] = {
+				title: titles[index] || index,
+				subtitle: filename,
+				type: Game.ItemTypes.Image,
+				fnExtract,
+				fnReplace,
+
+				// Function to open the file and return an Image instance.
+				fnOpen: () => {
+					const content = {
+						main: fnExtract(),
+					};
+
+					let imgData = img_raw_linear_8bpp.read(content, {
+						width: 320,
+						height: 200,
+					});
+
+					if (index in palettes) {
+						const palContent = {main: getFileFromDat(this.datTest, palettes[index]).getContent()};
+						const pal = pal_vga_6bit_papyrus.read(palContent);
+						imgData.palette = pal.palette;
+					}
+					
+					return imgData;
+				},
+				fnSave: (content) => {
+					const outContent = img_raw_linear_8bpp.write(content, {
+						width: 320,
+						height: 200,
+					});
+					fnReplace(outContent.content.main);
+
+					return {
+						warnings: outContent.warnings,
+					};
+				},
+			};
+		});
+
+		//
+		// Inventory object images
+		//
+		let inventitem = {};
+		const inventFilePatternAttr = 'inv%.4d.stp';
+		const numIventoryItems = 254;
+
+		for (let i = 1; i <= numIventoryItems; i += 1) {
+			
+			const numString = `${i}`;
+			const paddedNum = numString.padStart(4, '0');
+			const filename = inventFilePatternAttr.replace('%.4d', `${paddedNum}`);
```

Comment:
I wonder whether this could be simplified to something like:

    const paddedNum = i.toString().padStart(4, '0');
    const filename = `inv${paddedNum}.stp`;